src/ray/core_worker/actor_manager.cc

```diff
@@ -179,23 +179,10 @@ bool ActorManager::AddActorHandle(std::unique_ptr<ActorHandle> actor_handle,
   }
 
   if (inserted) {
-    // Register a callback to handle actor notifications.
-    auto actor_notification_callback =
-        std::bind(&ActorManager::HandleActorStateNotification,
-                  this,
-                  std::placeholders::_1,
-                  std::placeholders::_2);
-    RAY_CHECK_OK(gcs_client_->Actors().AsyncSubscribe(
-        actor_id,
-        actor_notification_callback,
-        [this, actor_id, cached_actor_name](Status status) {
-          if (status.ok() && !cached_actor_name.empty()) {
-            {
-              absl::MutexLock lock(&cache_mutex_);
-              cached_actor_name_to_ids_.emplace(cached_actor_name, actor_id);
-            }
-          }
-        }));
+    if (!cached_actor_name.empty()) {
+      absl::MutexLock lock(&cache_mutex_);
+      cached_actor_name_to_ids_.emplace(cached_actor_name, actor_id);
+    }
```

Comment:
@WangTaoTheTonic  I moved this logic from `on_done` callback to here directly to decouple the `ActorManager::SubscribeActorState`, please checking that if it is ok :)

Reply:
This doesn't seem right to me. Won't this only work if AddActorHandle gets called after the subscription finishes? Please add a unit test for this case to make sure there are no regressions.

It looks like you can regenerate the cached actor name from the actor handle when subscription completes so I'm not sure why it needs to be passed as an arg to this function anyway.

Reply:
> This doesn't seem right to me. Won't this only work if AddActorHandle gets called after the subscription finishes? Please add a unit test for this case to make sure there are no regressions.
> 
> It looks like you can regenerate the cached actor name from the actor handle when subscription completes so I'm not sure why it needs to be passed as an arg to this function anyway.

Thanks for reminding, I did not known that the actor name could be regenerated from the actor handle (Because we are still using version 1.2 internally, we can't regenerate it yet). It will simplify a lot of codes if the actor name could be regenerated :)